datafusion/expr/src/type_coercion/functions.rs

```diff
@@ -78,6 +78,9 @@ fn get_valid_types(
                 .map(|_| current_types[0].clone())
                 .collect()]
         }
+        TypeSignature::VariadicAny => vec![(0..current_types.len())
+            .map(|i| current_types[i].clone())
+            .collect()],
```

Comment:
Would the more simple `vec![current_types.iter().map(|c| c.clone()).collect()]` work too?